### PR TITLE
Added 2 unit tests for BasicLSTMCell to check ValueError.

### PR DESCRIPTION
**Pull request by chris-chris**
- Tests that dimension 0 in both(x and m) shape must be equal.
- Tests that state_size must be num_units * 2.

### CODE REVIEW

File: tensorflow/contrib/rnn/python/kernel_tests/core_rnn_cell_test.py [hunk 1]
```diff
@@ -194,6 +194,50 @@ def testBasicLSTMCell(self):
...
+          "root", initializer=init_ops.constant_initializer(0.5)):
+        num_units = 2
+        state_size = num_units * 3 # state_size must be num_units * 2
+        batch_size = 3
+        input_size = 4
+        x = array_ops.zeros(
+          [batch_size, input_size])
+        m = array_ops.zeros([batch_size, state_size])
+        with self.assertRaisesRegexp(ValueError,
+            r"Dimensions must be equal, but are 3 and 2"):
```
Comment (vrv) [inline tensorflow/contrib/rnn/python/kernel_tests/core_rnn_cell_test.py]: It becomes difficult to change error messages when tests check for exact contents, can we just do self.assertRaises instead?

Comment (chris-chris) [inline tensorflow/contrib/rnn/python/kernel_tests/core_rnn_cell_test.py]: @vrv Okay, I'll fix it! Thanks!

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Pull request merged**